lib/rubygems/package.rb

```diff
@@ -407,6 +407,31 @@ def gzip_to io # :yields: gz_io
     gz_io.close
   end
 
+  ##
+  # Returns file mode for installing +filename+
+
+  def install_mode filename, mode
+    if
+      mode & 0111 > 0 ||
+      file_is_executable(filename)
+    then
+      0775
```

Comment:
would be cool if we detected that this is `mode & 0111` but `@spec.bindir` OR `@spec.executables` were empty